src/utils/param-loader.ts

```diff
@@ -0,0 +1,81 @@
+import {CLI} from "./cli";
+import * as os from "os";
+
+
+export class ParamLoader {
+
+    private static instance : ParamLoader;
+
+    public static getInstance() : ParamLoader{
+        if (!ParamLoader.instance) {
+            ParamLoader.instance = new ParamLoader();
+        }
+        return ParamLoader.instance;
+    }
+
+    public all;
+    public CLI;
+    public config;
+    public env;
+    public DEFAULT ={
+        dbHost: "localhost",
+        dbPort: "5432",
+        dbDatabase: "mcm",
+        blobAccount: "wetstorage",
+        blobContainer: "websites",
+        processes: os.cpus().length,
+        crawlVersion: "CC-MAIN-2017-13",
+        heuristicThreshold: 3,
+        languageCodes: ["en"]
+    };
+
+    private constructor() {
+        this.CLI = CLI.getInstance().parameters;
+        this.config = ParamLoader.loadConfig();
+        this.env = ParamLoader.parseEnvVars();
+        this.all = Object.assign({}, this.DEFAULT, this.env, this.config, this.CLI);
```

Comment:
Exactly! The first parameter of `Object.assign` will be mutated. That's why there is an empty object.